libs/api/ebpf_api.cpp

```diff
@@ -1464,28 +1469,38 @@ _delete_ebpf_object(_In_opt_ _Post_invalid_ ebpf_object_t* object) noexcept
     }
 }
 
-static void
-_remove_ebpf_object_from_globals(_In_ const ebpf_object_t* object) noexcept
+_Requires_lock_not_held_(_ebpf_state_mutex) static void _remove_ebpf_object_from_globals(
+    _In_ const ebpf_object_t* object) noexcept
 {
     EBPF_LOG_ENTRY();
     ebpf_assert(object);
+    std::unique_lock lock(_ebpf_state_mutex);
     auto it = std::find(_ebpf_objects.begin(), _ebpf_objects.end(), object);
     ebpf_assert(it != _ebpf_objects.end());
     _ebpf_objects.erase(it);
```

Comment:
Should this code be more resilient? If `ebpf_assert` is debug only, then it doesn't seem to be a strong requirement for the `object` to already be in the collection. If it's possible not to be in it, the code should handle the case. If it's an absolute requirement, perhaps a non-debug-only (if it is debug only) assert should be here to "fail fast".